src/wp-includes/blocks/image.php

```diff
@@ -30,14 +31,278 @@ function render_block_core_image( $attributes, $content ) {
+	}
+
+	// Determine whether the view script should be enqueued or not.
+	$view_js_file = 'wp-block-image-view';
+	if ( ! wp_script_is( $view_js_file ) ) {
+		$script_handles = $block->block_type->view_script_handles;
+
+		// If the script is not needed, and it is still in the `view_script_handles`, remove it.
+		if ( ! $lightbox_enabled && in_array( $view_js_file, $script_handles, true ) ) {
+			$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
+		}
+		// If the script is needed, but it was previously removed, add it again.
+		if ( $lightbox_enabled && ! in_array( $view_js_file, $script_handles, true ) ) {
+			$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
+		}
+	}
+
+	if ( $lightbox_enabled ) {
+		return block_core_image_render_lightbox( $processor->get_updated_html(), $block->parsed_block );
+	}
+
 	return $processor->get_updated_html();
 }
 
+/**
+ * Add the lightboxEnabled flag to the block data.
+ *
+ * This is used to determine whether the lightbox should be rendered or not.
+ *
+ * @param  array $block Block data.
+ * @return array        Filtered block data.
+ */
+function block_core_image_get_lightbox_settings( $block ) {
+	// Get the lightbox setting from the block attributes.
+	if ( isset( $block['attrs']['lightbox'] ) ) {
+		$lightbox_settings = $block['attrs']['lightbox'];
+		// If the lightbox setting is not set in the block attributes,
+		// check the legacy lightbox settings that are set using the
+		// `gutenberg_should_render_lightbox` filter.
+		// We can remove this elseif statement when the legacy lightbox settings are removed.
+	} elseif ( isset( $block['legacyLightboxSettings'] ) ) {
+		$lightbox_settings = $block['legacyLightboxSettings'];
+	}
+
+	if ( ! isset( $lightbox_settings ) ) {
+		$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ), array( 'block_name' => 'core/image' ) );
+
+		// If not present in global settings, check the top-level global settings.
+		//
+		// NOTE: If no block-level settings are found, the previous call to
+		// `gutenberg_get_global_settings` will return the whole `theme.json`
+		// structure in which case we can check if the "lightbox" key is present at
+		// the top-level of the global settings and use its value.
+		if ( isset( $lightbox_settings['lightbox'] ) ) {
+			$lightbox_settings = gutenberg_get_global_settings( array( 'lightbox' ) );
+		}
+	}
+
+	return $lightbox_settings ?? null;
+}
+
+/**
+ * Add the directives and layout needed for the lightbox behavior.
+ *
+ * @param  string $block_content        Rendered block content.
+ * @param  array  $block                Block object.
+ * @return string                Filtered block content.
+ */
+function block_core_image_render_lightbox( $block_content, $block ) {
+	$processor = new WP_HTML_Tag_Processor( $block_content );
+
+	$aria_label = __( 'Enlarge image', 'gutenberg' );
```

Comment:
```suggestion
	$aria_label = __( 'Enlarge image' );
```

The Core did not use a text domain, and as a result, it should be removed from the PR. cc. @tellthemachines

Reply:
Oh well spotted! I'll add that to the Gutenberg PR.

Reply:
@anton-vlasenko, the usage of the `gutenberg` textdomain is another common issue when backporting from Gutenberg. We should prevent adding the textdomain for shared code in the first place.

@swissspidy, I know you coded ESLint rules preventing that in JavaScript, do you know if there is anything like that for PHP? One thing that I know for sure is that such a rule isn't enabled in WordPress core, because I had to apply a similar fix last week: https://github.com/WordPress/wordpress-develop/commit/fd894bfd2ccc057323907e65bb347fa81a35faa9.

Reply:
Yes. WPCS does allow specifying the text domain in tge config and can warn if a wrong one is used.

Reply:
@gziolo Thanks for the ping.
> We should prevent adding the textdomain for shared code in the first place.

I would like to clarify: do you mean this should be prevented in Gutenberg, or in Core?

Reply:
In the Gutenberg plugin, because the file gets copied to the WP core during the packages sync process.